src/merkle/empty_nodes.rs

```diff
@@ -0,0 +1,649 @@
+use super::{Felt, RpoDigest, WORD_SIZE, ZERO};
+use core::slice;
+
+// EMPTY NODES SUBTREES
+// ================================================================================================
+
+/// An empty sub-trees set.
+///
+/// Will provide precomputed node values for path computation for pre-defined depths.
+pub struct EmptyNodesSubtrees;
+
+impl EmptyNodesSubtrees {
+    // PROVIDERS
+    // --------------------------------------------------------------------------------------------
```

Comment:
minor nit: I'd probably remove this (since we have only one method here).

Reply:
Done!